WowPacketParser/Parsing/Parsers/UpdateHandler.cs

```diff
@@ -172,175 +180,336 @@ public static void ReadObjectsBlock(Packet packet, object index)
+                            packet.AddValue("(" + start + ") " + key, new WowGuid128(low, high), index);
+                        }
+                        break;
+                    }
+                    case UpdateFieldType.Quaternion:
+                    {
+                        packet.AddValue("(" + start + ") " + key, new Quaternion(fieldData[0].FloatValue, fieldData[1].FloatValue, fieldData[2].FloatValue, fieldData[3].FloatValue), index);
+                        break;
+                    }
+                    case UpdateFieldType.PackedQuaternion:
+                    {
+                        long quat = fieldData[1].UInt32Value;
+                        quat <<= 32;
+                        quat += fieldData[0].UInt32Value;
+                        packet.AddValue("(" + start + ") " + key, new Quaternion(quat), index);
+                        break;
+                    }
+                    case UpdateFieldType.Uint:
+                    {
+                        for (int k = 0; k < fieldData.Count; ++k)
+                            if (mask[start + k])
+                                packet.AddValue("(" + (start + k) + ") " + (k > 0 ? key + " + " + k : key), fieldData[k].UInt32Value, index);
+                        break;
+                    }
+                    case UpdateFieldType.Int:
+                    {
+                        for (int k = 0; k < fieldData.Count; ++k)
+                            if (mask[start + k])
+                                packet.AddValue("(" + (start + k) + ") " + (k > 0 ? key + " + " + k : key), fieldData[k].Int32Value, index);
+                        break;
+                    }
+                    case UpdateFieldType.Float:
+                    {
+                        for (int k = 0; k < fieldData.Count; ++k)
+                            if (mask[start + k])
+                                packet.AddValue("(" + (start + k) + ") " + (k > 0 ? key + " + " + k : key), fieldData[k].FloatValue, index);
+                        break;
+                    }
+                    case UpdateFieldType.Bytes:
+                    {
+                        for (int k = 0; k < fieldData.Count; ++k)
+                        {
+                            if (mask[start + k])
+                            {
+                                byte[] intBytes = BitConverter.GetBytes(fieldData[k].UInt32Value);
+                                if (BitConverter.IsLittleEndian)
+                                    Array.Reverse(intBytes);
+                                byte[] result = intBytes;
+                                packet.AddValue("(" + (start + k) + ") " + (k > 0 ? key + " + " + k : key), result[0] + "/" + result[1] + "/" + result[2] + "/" + result[3], index);
+                            }
+                        }
+                        break;
+                    }
+                    case UpdateFieldType.Custom:
```

Comment:
Again, I don't like splitting implementation from definition here, especially for corner cases/hacks, but besides shoving a delegate in the attribute, idk

Reply:
There will be plenty, for example parsing of unitflags, or parsing of bytes 0, bytes1 and so on, for which a delegate should be provided based on the version, but at the moment I did not want to spend time on it, since this on its own is a huge improvement already.